server/db/driver.go

```diff
@@ -1,24 +1,56 @@
 package db
 
-// var (
-// 	driverMtx  sync.Mutex
-// 	driverName string
-// 	driver     Driver
-// )
+import (
+	"context"
+	"fmt"
+	"sync"
 
-// type Driver interface {
-// 	Open(ctx context.Context, cfg interface{}) (DEXArchivist, error)
-// }
+	"decred.org/dcrdex/dex"
+)
 
-// func Register(name string, d Driver) {
-// 	driverMtx.Lock()
-// 	defer driverMtx.Unlock()
-// 	if d == nil {
-// 		panic("db: Register driver is nil")
-// 	}
-// 	if driver != nil {
-// 		panic("db: Register already called. driver: " + driverName)
-// 	}
-// 	driverName = name
-// 	driver = d
-// }
+var (
+	driverMtx  sync.Mutex
+	driverName string
+	driver     Driver
+)
+
+// Driver is the interface required of all DB drivers. Open should create a
+// DEXArchivist and verify connectivity with the asset's chain server.
+type Driver interface {
+	Open(ctx context.Context, cfg interface{}) (DEXArchivist, error)
+	UseLogger(logger dex.Logger)
+}
+
+// Register should be called by the init function of an DB driver's package.
+func Register(name string, d Driver) {
+	driverMtx.Lock()
+	defer driverMtx.Unlock()
+	if d == nil {
+		panic("db: Register driver is nil")
+	}
+	if driver != nil {
+		panic(fmt.Sprintf("db: Register already called. driver: %q", driverName))
+	}
+	driverName = name
+	driver = d
+}
```

Comment:
I did it this way because unlike the asset backends where it makes sense to have several loaded at once, you would never actually open more than one DB driver.  However, that's just for dcrdex and perhaps there would be need to have more than one loaded (backend migration perhaps).  Changed.